addons/hr_holidays/models/hr_leave_type.py

```diff
@@ -290,13 +290,14 @@ def _search_virtual_remaining_leaves(self, operator, value):
     @api.depends_context('employee_id', 'default_employee_id', 'default_date_from')
     def _compute_leaves(self):
         employee = self.env['hr.employee']._get_contextual_employee()
-        target_date = self._context['default_date_from'] if 'default_date_from' in self._context else None
+        target_date = self.env.context.get('default_date_from')
+        # TODO remove this HACK
         # This is a workaround to save the date value in context for next triggers
         # when context gets cleaned and 'default_' context keys gets removed
         if target_date:
-            self.env.context = frozendict(self.env.context, leave_date_from=self._context['default_date_from'])
+            vars(self.env)['context'] = frozendict(self.env.context, leave_date_from=self._context['default_date_from'])
```

Comment:
IMO, we should find a better solution for the bug that introduced this fix.

Reply:
This part is quite messy and I did not want to refactor too much inside this PR, I left a TODO comment.

Reply:
I don't see the point of making `env` readonly if you keep this mess.  That's clearly a "how to get around it"...  I guess that the modified context is useful for method `get_allocation_data()`.  If so, add a `with_context()` where needed, and call it a day.

Reply:
I have removed the hack for now. Only one test covers this. We call `clean_context` when creating a new environment. It means that `self.with_context(default_date_from=...).sudo().display_name` does not have the default set and computes the value incorrectly.
It is not as simple as adding a few with_context. The obvious fix is to stop using any "defualt_*" in the `depends_context` - we should probably raise a warning for that case - and use some other key that is not prefixed with "default_".

What do you think is the bast way to go forward?

Reply:
@Bertrand2 Could you review the change?